internal/batches/docker/image.go

```diff
@@ -66,19 +68,50 @@ func (image *image) Ensure(ctx context.Context) error {
 	image.ensureOnce.Do(func() {
 		image.ensureErr = func() (err error) {
 			inspectDigest := func() (string, error) {
-				// TODO!(sqs): is image id the right thing to use here? it is NOT
-				// the digest. but the digest is not calculated for all images
-				// (unless they are pulled/pushed from/to a registry), see
-				// https://github.com/moby/moby/issues/32016.
```

Comment:
This `TODO` was outdated: there's actually a lengthy comment earlier in the file explaining what "digest" means in this context, and how it relates to the public digest. Whoever last touched this really should have removed this then. 😬